test/base-tests/upload-to-aws.sh

```diff
@@ -0,0 +1,151 @@
+if [[ $ID == rhel ]]; then
+    sudo sed -r -i 's#toml.load\(args\[3\]\)#toml.load(open(args[3]))#' \
+        /usr/lib/python3.6/site-packages/composer/cli/compose.py
+    sudo rm -f /usr/lib/python3.6/site-packages/composer/cli/compose.pyc
+fi
+
+# We need jq for parsing composer-cli output.
+if ! hash jq; then
+    greenprint "Installing jq"
+    sudo dnf -qy install jq
+fi
+
+TEST_ID="${CHANGE_ID}-${BUILD_ID}"
+IMAGE_KEY=osbuild-composer-base-test-${TEST_ID}
+
+# Jenkins sets WORKSPACE to the job workspace, but if this script runs
+# outside of Jenkins, we can set up a temporary directory instead.
+if [[ ${WORKSPACE:-empty} == empty ]]; then
+    WORKSPACE=$(mktemp -d)
+fi
+
+# Set up temporary files.
+TEMPDIR=$(mktemp -d)
+AWS_CONFIG=${TEMPDIR}/aws.toml
+BLUEPRINT_FILE=${TEMPDIR}/blueprint.toml
+BLUEPRINT_NAME=targetimage
+COMPOSE_START=${TEMPDIR}/compose-start-${IMAGE_KEY}.json
+COMPOSE_INFO=${TEMPDIR}/compose-info-${IMAGE_KEY}.json
+
+# Get the compose log.
+get_compose_log () {
+    COMPOSE_ID=$1
+    LOG_FILE=${WORKSPACE}/osbuild-${ID}-${VERSION_ID}-aws.log
+
+    # Download the logs.
+    sudo composer-cli compose log $COMPOSE_ID | tee $LOG_FILE > /dev/null
+}
+
+# Get the compose metadata.
+get_compose_metadata () {
+    COMPOSE_ID=$1
+    METADATA_FILE=${WORKSPACE}/osbuild-${ID}-${VERSION_ID}-aws.json
+
+    # Download the metadata.
+    sudo composer-cli compose metadata $COMPOSE_ID > /dev/null
+
+    # Find the tarball and extract it.
+    TARBALL=$(basename $(find . -maxdepth 1 -type f -name "*-metadata.tar"))
+    tar -xf $TARBALL
+    rm -f $TARBALL
+
+    # Move the JSON file into place.
+    cat ${COMPOSE_ID}.json | jq -M '.' | tee $METADATA_FILE > /dev/null
+}
+
+# Write an AWS TOML file
+tee $AWS_CONFIG > /dev/null << EOF
+provider = "aws"
+
+[settings]
+accessKeyID = "${AWS_ACCESS_KEY_ID}"
+secretAccessKey = "${AWS_SECRET_ACCESS_KEY}"
+bucket = "${AWS_BUCKET}"
+region = "${AWS_REGION}"
+key = "${IMAGE_KEY}"
+EOF
+
+# Write a basic blueprint for our image.
+tee $BLUEPRINT_FILE << EOF
+name = "${BLUEPRINT_NAME}"
+description = "A base system with bash"
+version = "0.0.1"
+
+[[packages]]
+name = "bash"
+
+[[packages]]
+name = "osbuild-composer"
+
+[customizations.services]
+enabled = ["sshd", "cloud-init", "cloud-init-local", "cloud-config", "cloud-final", "osbuild-composer.socket"]
+EOF
+
+# Add osbuild-mock repo as a source to osbuild-composer
+greenprint "📋 Adding sources"
+sudo composer-cli sources add osbuild-mock.toml
+
+sudo composer-cli sources list
+sudo composer-cli sources info osbuild-mock
+
+# Prepare the blueprint for the compose.
+greenprint "📋 Preparing blueprint"
+cat $BLUEPRINT_FILE
+sudo composer-cli -j blueprints push $BLUEPRINT_FILE
+greenprint "📋 Depsolving blueprint"
+sudo composer-cli -j blueprints depsolve $BLUEPRINT_NAME
+
+# Get worker unit file so we can watch the journal.
+WORKER_UNIT=$(sudo systemctl list-units | egrep -o "osbuild.*worker.*\.service")
+sudo journalctl -af -n 1 -u ${WORKER_UNIT} &
```

Comment:
Umpf. This will cause random log entries in the output. Preferably the action that waits does this, but I guess that is not how composer-cli works.